src/query/storages/fuse/fuse/src/operations/mutation/compact/compact_transform.rs

```diff
@@ -143,18 +184,96 @@ impl Processor for CompactTransform {
             return Ok(Event::NeedData);
         }
 
-        self.state = State::Compact(self.input.pull_data().unwrap()?);
+        let input_date = self.input.pull_data().unwrap()?;
+        let meta = input_date.get_meta().unwrap();
+        let task_meta = CompactSourceMeta::from_meta(meta)?;
+        self.order = task_meta.order;
+        self.compact_tasks = task_meta.tasks.clone();
+
+        self.state = State::ReadBlocks;
         Ok(Event::Async)
     }
 
     fn process(&mut self) -> Result<()> {
```

Comment:
Could you add more docs for the parallel compact algorithm?
It's better to read the doc than the code.